analytics.py

```diff
@@ -0,0 +1,234 @@
+from .utils import *
+
+def find_largest_city(gj):
+    """
+    Iterate through a geojson feature collection and
+    find the largest city.  Assume that the key
+    to access the maximum population is 'pop_max'.
+
+    Parameters
+    ----------
+    gj : dict
+         A GeoJSON file read in as a Python dictionary
+
+    Returns
+    -------
+    city : str
+           The largest city
+
+    population : int
+                 The population of the largest city
+    """
+    city = None
+    max_population = 0
+
+    features = gj['features']
+    for i in features:
+        if(i['properties']['pop_max']>max_population):
+            max_population = i['properties']['pop_max']
+            city = i['properties']['name']
+
+    return city, max_population
+
+def write_your_own(gj):
+    """
+    Here you will write your own code to find
+    some attribute in the supplied geojson file.
+
+    Take a look at the attributes available and pick
+    something interesting that you might like to find
+    or summarize.  This is totally up to you.
+
+    Do not forget to write the accompanying test in
+    tests.py!
+    """
+
+    #I chose the least creative route of finding the average population among all cities
+
+    sum_population = 0
+    n = 0
+
+    features = gj['features']
+    for i in features:
+        sum_population = i['properties']['pop_max']+sum_population
+        n = n+1
+
+    return sum_population/n
+
+def mean_center(points):
+    """
+    Given a set of points, compute the mean center
+
+    Parameters
+    ----------
+    points : list
+         A list of points in the form (x,y)
+
+    Returns
+    -------
+    x : float
+        Mean x coordinate
+
+    y : float
+        Mean y coordinate
+    """
+    x = 0
+    y = 0
+    n = 0
+
+    for i in points:
+        x = x+i[0]
+        y = y+i[1]
+        n = n+1
+
+    x = x/n
+    y = y/n
+    return x, y
+
+
+# def average_nearest_neighbor_distance(points):
```

Comment:
You can use triple quotes at the start and stop of a code block to comment it out.

```python
"""
x = 1
y = 2
z = 3
"""
```
All of the above is commented out, without having to insert many #.